Tasks/Zhukovets/Task2/Task2.py

```diff
@@ -0,0 +1,117 @@
+        return variant3
+
+
+def time_convertor(time_string):
+    time_obj = datetime.datetime.strptime(time_string, "%H:%M").time()
+    hour = time_obj.hour if time_obj.hour < 12 else time_obj.hour - 12
+    minutes = time_obj.minute
+    if time_obj.hour == 0 and time_obj.minute == 0:
+        return 'Сейчас полночь'
+    elif time_obj.hour == 12 and time_obj.minute == 0:
+        return 'Сейчас полдень'
+    elif minutes == 0:
+        value_choose_plural = choose_plural(hour, "час", "часа", "часов")
+        value_from_dict_for_hours = digit[hour][0]
+        return f'Сейчас {value_from_dict_for_hours} {value_choose_plural} ровно'
+
+    elif minutes == 30:
+        value_from_dict_for_hours = digit[hour+1][3]
+        return f'Половина {value_from_dict_for_hours}'
+
+    elif minutes < 30:
+        value_choose_plural = choose_plural(minutes, "минута", "минуты", "минут")
+        value_from_dict_for_hours = digit[hour + 1][3]
+        if minutes <= 20:
+            value_from_dict_for_minutes = digit[minutes][2]
+        else:
+            remainder = time_obj.minute - 20
+            value_from_dict_for_minutes = f'{tens[20]} {digit[remainder][2]}'
+        return f'{value_from_dict_for_minutes} {value_choose_plural} {value_from_dict_for_hours}'
+
+    elif 30 < minutes < 45:
+        value_choose_plural = choose_plural(minutes, 'минута', "минуты", "минут")
+        value_from_dict_for_hours = digit[hour + 1][3]
+        if minutes < 40:
+            remainder = minutes - 30
+            value_from_dict_for_minutes = f'{tens[30]} {digit[remainder][2]}'
+        else:                       # minutes from 40 to 45
+            remainder = minutes - 40
+            value_from_dict_for_minutes = f'{tens[40]} {digit[remainder][2]}'
+        return f'{value_from_dict_for_minutes} {value_choose_plural} {value_from_dict_for_hours}'
+
+    else:                           # minutes <= 45
+        value_choose_plural = choose_plural(hour + 1, "час", "часа", "часов")
+        value_from_dict_for_hours = digit[hour+1][0]
+        value_from_dict_for_minutes = digit[60 - minutes][1]
+        str_minutes = "минут" if minutes != 1 else "минуты"
+        return f'Без {value_from_dict_for_minutes} {str_minutes} {value_from_dict_for_hours} {value_choose_plural}'
+
+
+def main():
+    while True:
+        user_choice = input("Какая опция вас интересует?\n"
+                            "Если вы хотите узнать текущее время - нажмите '1'\n"
+                            "Если вы хотите задать время - нажмите '2'\n"
+                            ":")
+        if user_choice == '1':
+            time_string = datetime.datetime.now().strftime("%H:%M")
+            return time_convertor(time_string)
+        elif user_choice == '2':
+            try:
+                time_string = input("Пожалуйста введите интересующее вас время в формате 'ЧЧ:ММ'\n: ")
+                validation_value1, validation_value2 = time_string.split(":")[0], time_string.split(":")[-1]
```

Comment:
and what if there is no ":" symbol in the string? you should've check it beforehand, -1 point for the unwise choice